core/commands/storage/contracts/contracts.go

```diff
@@ -370,6 +374,33 @@ func SyncContracts(ctx context.Context, n *core.IpfsNode, req *cmds.Request, env
 		if err != nil {
 			return err
 		}
+
+		go func() {
+			if b := ctx.Value(contractsSyncVerboseOptionName); b != nil && b.(bool) {
```

Comment:
Put this check outside the goroutine. Because ctx may be gone if the main routine is too fast.